cmd/orch-tester/broadcaster_metrics.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"go.opencensus.io/stats/view"
+	"sync"
+)
+
+type broadcasterMetrics struct {
+	// Distribution metrics grouped by metric name
+	lastSum   map[string]float64
+	lastCount map[string]int
+	rateSum   map[string]float64
+	rateCount map[string]int
+
+	// Error count metrics by grouped by error_code tag
+	lastErrs map[string]int
+	incErrs  map[string]int
+
+	mu sync.Mutex
+}
+
+func (bm *broadcasterMetrics) reset() {
+	bm.mu.Lock()
+	defer bm.mu.Unlock()
+
+	bm.rateSum = map[string]float64{}
+	bm.rateCount = map[string]int{}
+	bm.incErrs = map[string]int{}
+}
+
+func (bm broadcasterMetrics) ExportView(viewData *view.Data) {
+	bm.mu.Lock()
+	defer bm.mu.Unlock()
+
+	name := viewData.View.Name
+	rows := viewData.Rows
+	if len(rows) <= 0 {
+		return
+	}
+
+	bm.handleDistributionMetrics(rows, name)
+	bm.handleErrorCountMetrics(rows, name)
+}
+
+func (bm broadcasterMetrics) handleDistributionMetrics(rows []*view.Row, name string) {
+	supportedMetrics := map[string]bool{
```

Comment:
Why isn't this a package variable?

Reply:
It could be, but I think it's better to limit the scope to the function if it's only used inside that function.